core/server/middleware/middleware.js

```diff
@@ -76,6 +81,16 @@ function sslForbiddenOrRedirect(opt) {
     return response;
 }
 
+function verifySessionHash(hash) {
+    var bcryptCompare = Promise.promisify(bcrypt.compare);
+    return api.settings.read({context: {internal: true}, key: 'password'}).then(function (response) {
+        var pass = response.settings[0].value;
+        return bcryptCompare(pass, hash).then(function (verified) {
+            return verified;
+        });
```

Comment:
No need to use an explicit `then` in this case--you can simply `return bcryptCompare(pass, hash);` to the caller.